### ISSUE TITLE
Can it import a firmware?

### ISSUE THREAD

**Issue by yiivon** [wokwi/wokwi-features#78]
That would be more wonderful if wokwi-playgrounds can importing a firmware and run it directly, it possibles?

**Comment by urish** [comment]
Hi Yiivon,

It's now possible to import a HEX file:

1. Click on the code editor and press "F1"
2. Select "Load HEX File and Start Simulation..."
3. Choose a HEX file to load

![image](https://user-images.githubusercontent.com/892318/117701801-54dcc300-b1d0-11eb-852c-aa99f5b65f0f.png)

Would love to hear your feedback about this new feature!

**Comment by guymcswain** [comment]
Is this feature still available - "Load HEX File and Start Simulation..."
I'm not seeing it in the code editor on my browser (MS Edge nor Google Chrome).

**Comment by urish** [comment]
The feature is indeed still available. What do you see when you press F1 inside the code editor?

**Comment by guymcswain** [comment]
I get a drop down list: "Add cursor above", "add cursor below", etc, but not this particular item.  I'm a noob with wokwi.

**Comment by urish** [comment]
That's ok, it's a good start. Then, what happens when you type the word "HEX" inside this drop down? what options do you see then?

**Comment by guymcswain** [comment]
That is it.  I see it now.  Thank you!

**Comment by urish** [comment]
You are welcome :)

**Comment by green-mcu** [comment]
**urish**, is it possible to store HEX on the server?

**Comment by urish** [comment]
On Wokwi's server? What's your use case for that?

**Comment by green-mcu** [comment]
Yes, on Wokwi's server. So that anyone can do HEX without a sketch.

**Comment by green-mcu** [comment]
So that everyone can run the program without having the source code.

**Comment by urish** [comment]
Actually, we do want to encourage people to share their source code.

An alternative for people who want to share the hex without the source code is to put the files on GitHub, where they can be viewed using Wokwi for VSCode. Here is an example for such setup: https://github.com/wokwi/esp-idf-hello-world

**Comment by green-mcu** [comment]
OK, **urish**. Thanks.

**Comment by hayschan** [comment]
> So that everyone can run the program without having the source code.

This is also a use case for me. I would like to do training for my end users without giving them the source code nor the `.hex` file. Wokwi will be a brilliant platform for that.

> An alternative for people who want to share the hex without the source code is to put the files on GitHub, where they can be viewed using Wokwi for VSCode.

Using the browser is always the first option for non-technical users. Will be great if that is achievable.

**Comment by urish** [comment]
> This is also a use case for me. I would like to do training for my end users without giving them the source code nor the .hex file. Wokwi will be a brilliant platform for that.

Thanks for explaining your use case! If we launched that as a paid feature, would that work for you?

**Comment by hayschan** [comment]
> If we launched that as a paid feature, would that work for you?

Yep, I have no problem for paying. I understand advanced feature like these deserve being in the subscription plan.
